src/main/kotlin/nl/nuts/discovery/store/entity/PartyId.kt

```diff
@@ -0,0 +1,42 @@
+package nl.nuts.discovery.store.entity
+
+import org.bouncycastle.asn1.x509.OtherName
+import javax.persistence.AttributeConverter
+import javax.persistence.Convert
+
+
+data class PartyId(val oid: String, val value: String) {
+
+    companion object {
+        fun parse(fullyQualified: String): PartyId {
+            val parts = fullyQualified.split(":")
+            if (parts.size != 4) {
+                throw IllegalArgumentException("Invalid number of parts in OID URN: $fullyQualified")
+            }
+            if (parts[0] != "urn" || parts[1] != "oid") {
```

Comment:
requiring `oid` is very restrictive. We have to remember this.....

Reply:
Party identifiers (subject/custodian/actor) are always in the form of `urn:oid:<oid>:<value>`, as specified (https://nuts-documentation.readthedocs.io/en/latest/pages/technical/identifiers.html). So I regard the strictness of requiring it to be in that exact format as an improvement over the current situation (free format).